src/test/java/io/cdap/plugin/servicenow/sink/ServiceNowSinkConfigTest.java

```diff
@@ -0,0 +1,411 @@
+    List<Map<String, Object>> result = new ArrayList<>();
+    Map<String, Object> map = new HashMap<>();
+    map.put("key", "value");
+    result.add(map);
+    int httpStatus = HttpStatus.SC_NOT_FOUND;
+    Map<String, String> headers = new HashMap<>();
+    String responseBody = "{\n" +
+      "    \"result\": [\n" +
+      "        {\n" +
+      "            \"calendar_integration\": \"1\",\n" +
+      "            \"country\": \"\",\n" +
+      "            \"date_format\": \"\",\n" +
+      "            \"location\": \"\"\n" +
+      "        }\n" +
+      "    ]\n" +
+      "}";
+    SchemaResponse schemaResponse = new SchemaResponse();
+    schemaResponse.setLabel("Class");
+    schemaResponse.setName("sys_class_name");
+    schemaResponse.setInternalType("sys_class_name");
+    schemaResponse.setExampleValue("");
+    RestAPIResponse restAPIResponse = new RestAPIResponse(httpStatus, headers, responseBody);
+    OAuthClient oAuthClient = Mockito.mock(OAuthClient.class);
+    PowerMockito.whenNew(OAuthClient.class).
+      withArguments(Mockito.any(URLConnectionClient.class)).thenReturn(oAuthClient);
+    OAuthJSONAccessTokenResponse accessTokenResponse = Mockito.mock(OAuthJSONAccessTokenResponse.class);
+    Mockito.when(oAuthClient.accessToken(Mockito.any(), Mockito.anyString(), Mockito.any(Class.class))).
+      thenReturn(accessTokenResponse);
+    Mockito.when(accessTokenResponse.getAccessToken()).thenReturn("token");
+    RestAPIResponse response = Mockito.spy(restAPIResponse);
+    CloseableHttpClient httpClient = Mockito.mock(CloseableHttpClient.class);
+    HttpClientBuilder httpClientBuilder = Mockito.mock(HttpClientBuilder.class);
+    PowerMockito.mockStatic(HttpClientBuilder.class);
+    PowerMockito.mockStatic(RestAPIResponse.class);
+    PowerMockito.when(HttpClientBuilder.create()).thenReturn(httpClientBuilder);
+    Mockito.when(httpClientBuilder.build()).thenReturn(httpClient);
+    CloseableHttpResponse httpResponse = Mockito.mock(CloseableHttpResponse.class);
+    Mockito.when(httpClient.execute(Mockito.any())).thenReturn(httpResponse);
+    PowerMockito.when(RestAPIResponse.parse(httpResponse, null)).thenReturn(response);
+    Mockito.when(restApi.executeGet(Mockito.any(RestAPIRequest.class))).thenReturn(restAPIResponse);
+    Mockito.when(restApi.fetchServiceNowTableSchema(Mockito.anyString(), Mockito.any())).thenReturn(schema);
+    Mockito.when(restApi.parseSchemaResponse(restAPIResponse.getResponseBody()))
+      .thenReturn(Collections.singletonList(schemaResponse));
+    try {
+      config.validateSchema(schema, collector);
+      collector.getOrThrowException();
+      Assert.fail("Exception is not thrown if operation is provided");
+    } catch (ValidationException e) {
+      Assert.assertEquals("Errors were encountered during validation. Unable to fetch schema for table " +
+                            "tableName", e.getMessage());
+    }
+  }
+
+  @Test(expected = IllegalArgumentException.class)
```

Comment:
why is IllegalArgumentException expected here? Please add some comments

Reply:
Modified the testCase